### PR TITLE
feat: campaign group management ui

### PR DESCRIPTION
**Pull request by jeffersonrabb**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds a "Groups" tab for managing Campaign Groups with basic functionality. Groups can be created and deleted here. Some ideas for future iterations on this screen:

- Ability to rename groups
- Ability to Publish/Unpublish all campaigns in the group (redundant to Campaigns tab?)
- Ability to Preview

### How to test the changes in this Pull Request:

1. Observe new Groups tab in Campaigns wizard.
2. Try creating and deleting groups here.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by jeffersonrabb** [comment]
@thomasguillot could you give this a look and see if you have any UI suggestions?

File: assets/wizards/popups/views/campaign-group-management/index.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+				slug: term,
+			},
+		} ).then( () => {
+			setInFlight( false );
+			retrieveTerms();
+		} );
+	};
+
+	const deleteTerm = id => {
+		apiFetch( {
```
Comment (adekbadek) [inline assets/wizards/popups/views/campaign-group-management/index.js:47]: Suggestion: set `inFlight` to block UI when the request is made.

**Review by adekbadek — APPROVED** [review]

File: assets/wizards/popups/views/campaign-group-management/index.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+					{ __( 'Add New', 'newspack' ) }
+				</Button>
+				{ addNewPopoverIsVisible && (
+					<div className="newspack-campaigns-wizard-groups__add-new-button">
+						<Popover
+							position="bottom left"
+							onFocusOutside={ () => setAddNewPopoverIsVisible( false ) }
+							onKeyDown={ event => ESCAPE === event.keyCode && setAddNewPopoverIsVisible( false ) }
+						>
+							<MenuItem
```
Comment (jeffersonrabb) [inline assets/wizards/popups/views/campaign-group-management/index.js:125]: What's this `MenuItem` here for? I don't see the button and lint is complaining that `setPreviewPopoverIsVisible` isn't a real function.

File: assets/wizards/popups/views/campaign-group-management/index.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+	ActionCard,
+	Button,
+	Notice,
+	Popover,
+	TextControl,
+} from '../../../../components/src';
+import './style.scss';
+
+/**
+ * Popup group screen.
```
Comment (jeffersonrabb) [inline assets/wizards/popups/views/campaign-group-management/index.js:32]:
```suggestion
 * Campaign Groups management screen.
```

**Review by jeffersonrabb — COMMENTED** [review]
Left two little notes. Two other thoughts, which we could easily punt to a future PR:

- It'd be nice if the `TextControl` listened for the RETURN character and submitted when pressed.
- Eventually it'd be nice if `inFlight` were managed by the `withWizardScreen` HoC. `inFlight` would be available as a param and it would be true whenever an API request is in flight. This would allow disabling certain UI without having to create a new state variable each time. Just a thought, definitely not a blocker.

File: assets/wizards/popups/views/campaign-group-management/index.js (hunk 2 above)
Comment (dkoo) [reply assets/wizards/popups/views/campaign-group-management/index.js:125]: The `MenuItem` is a hidden element to provide a way for screen reader users to close the popover. This exists in the other popovers we use in the Campaigns tab as well.

Comment (jeffersonrabb) [reply assets/wizards/popups/views/campaign-group-management/index.js:125]: Got it, then the only change is the click event should be `setAddNewPopoverIsVisible( false )`

Comment (dkoo) [reply assets/wizards/popups/views/campaign-group-management/index.js:125]: Whoops, fixed in 5fbb5d3.

**Comment by dkoo** [comment]
> It'd be nice if the `TextControl` listened for the RETURN character and submitted when pressed.
Added in 3a0d6e4. This also automatically closes the popover when adding a new term.

> Eventually it'd be nice if `inFlight` were managed by the `withWizardScreen` HoC. `inFlight` would be available as a param and it would be true whenever an API request is in flight. This would allow disabling certain UI without having to create a new state variable each time. Just a thought, definitely not a blocker.
This is a great idea, but yes, a good candidate for a separate refactor PR as it touches a lot of different components. Added a new issue here: https://github.com/Automattic/newspack-plugin/issues/825

File: assets/wizards/popups/views/campaign-group-management/index.js (hunk 1 above)
Comment (dkoo) [reply assets/wizards/popups/views/campaign-group-management/index.js]: Done in 8baca9e

**Comment by dkoo** [comment]
Noting that @jeffersonrabb approved this offline (he couldn't officially approve since he opened it). Merging.

**Pull request merged**